src/client.ts

```diff
@@ -3121,7 +3122,7 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         opts: IKeyBackupRestoreOpts,
     ): Promise<IKeyBackupRestoreResult> {
         const privKey = decodeRecoveryKey(recoveryKey);
-        return this.restoreKeyBackup(privKey, targetRoomId, targetSessionId, backupInfo, opts);
+        return this.restoreKeyBackup(privKey, targetRoomId!, targetSessionId!, backupInfo, opts);
```

Comment:
I'm confused why we're saying that `targetRoomId` and `targetSessionId` are defined, aren't we explicitly allowing them to be undefined in the function's type signature and in the documentation?

Reply:
Because of the complex method signature overload, otherwise we'd need to have 3 if-cases (one for each overload) to be able to call it without error